bokehjs/src/coffee/model.coffee

```diff
@@ -17,6 +21,42 @@ export class Model extends HasProps
       for cb in callbacks
         @listenTo(@, evt, () -> cb.execute(@))
 
+    @listenTo(@, 'change:js_event_callbacks', () -> @_update_event_callbacks)
+
+
+  _process_event : (event) ->
+    # Given an applicable event, execute the associated callback
+    if not event.constructor.event_name
+       logger.warn('Event is still passed as a string: ' + event)
+       return
+    if not event.constructor.applicable_models.some((m) => m == @.type)
```

Comment:
If I import the various models into ``bokeh_events.ts`` then I could set the actual model classes in ``applicable_models``. Then I could use an isinstance check here. I'm happy to do that if I am understand it right.